### PR TITLE
 add nil check and add ut

### PR DESCRIPTION
**Pull request by MXuDong**
When getting the POD Annotation, there is no null pointer detection, add this detection, and add unit test on the whole function.

[x] Configuration Infrastructure
[ ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[x] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

Pull Request Attributes

Please check any characteristics that apply to this pull request. 

[x] Does not have any [user-facing](https://github.com/istio/istio/tree/master/releasenotes#when-to-add-release-notes) changes. This may include CLI changes, API changes, behavior changes, performance improvements, etc.

### CODE REVIEW

File: istioctl/cmd/injector-list.go [hunk 1]
```diff
@@ -303,6 +303,9 @@ func hideFromOutput(ns resource.Namespace) bool {
 }
 
 func injectionDisabled(pod *v1.Pod) bool {
+	if pod.ObjectMeta.GetAnnotations() == nil{
```
Comment (howardjohn) [inline istioctl/cmd/injector-list.go:306]: why do we need this? A nil map is valid

File: istioctl/cmd/injector-list_test.go [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+import (
+	"istio.io/api/annotation"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"testing"
+)
+
+func Test_injectionDisabled(t *testing.T) {
+	type args struct {
+		pod *v1.Pod
```
Comment (howardjohn) [inline istioctl/cmd/injector-list_test.go:26]: Not sure we need this struct since there is only one arg

File: istioctl/cmd/injector-list.go (hunk 1 above)
Comment (MXuDong) [reply istioctl/cmd/injector-list.go:306]: I thought something would go wrong with the null pointer, but it seems I was wrong。🤣

File: istioctl/cmd/injector-list_test.go (hunk 2 above)
Comment (MXuDong) [reply istioctl/cmd/injector-list_test.go:26]: What should I do?

**Comment by ericvn** [comment]
The release-notes check is failing as your branch `MXuDong:add_nil_check_for_pod_annotation` shows`This branch is 1 commit ahead, 1018 commits behind istio:master.`.  Because your PR is based on such an old commit, when the release notes code is run trying to find differences in release notes files, it includes all the commits and picks up that file was changed so tries to validate it. If you could rebase your changes on a more current istio/istio commit, the script shouldn't have to diff so many commits and the file won't show up.

@brian-avery, would it make sense to just not do any checks in the case the label is set.

**Comment by MXuDong** [comment]
> The release-notes check is failing as your branch `MXuDong:add_nil_check_for_pod_annotation` shows`This branch is 1 commit ahead, 1018 commits behind istio:master.`. Because your PR is based on such an old commit, when the release notes code is run trying to find differences in release notes files, it includes all the commits and picks up that file was changed so tries to validate it. If you could rebase your changes on a more current istio/istio commit, the script shouldn't have to diff so many commits and the file won't show up.
> 
> @brian-avery, would it make sense to just not do any checks in the case the label is set.

Oh, thanks, I decided to close this PR and resubmit.

**Pull request closed without merging**